Source/SG/SG.cs

```diff
@@ -98,22 +118,43 @@ public static PAENConfiguration GetDefault()
         public static void ConfigureInstrument(NIRfsg rfsgHandle, InstrumentConfiguration instrConfig)
         {
             rfsgHandle.SignalPath.SelectedPorts = instrConfig.SelectedPorts;
-            rfsgHandle.RF.ExternalGain = -instrConfig.ExternalAttenuation_dBm;
+            rfsgHandle.FrequencyReference.Source = RfsgFrequencyReferenceSource.FromString(instrConfig.ReferenceClockSource);
             rfsgHandle.RF.Configure(instrConfig.CarrierFrequency_Hz, instrConfig.DutAverageInputPower_dBm);
+            rfsgHandle.RF.ExternalGain = -instrConfig.ExternalAttenuation_dBm;
 
-            rfsgHandle.FrequencyReference.Source = RfsgFrequencyReferenceSource.FromString(instrConfig.ReferenceClockSource);
-            
-            if (instrConfig.ShareLOSGToSA)
+            foreach (LocalOscillatorConfiguration loConfig in instrConfig.LoConfigurations)
             {
-                rfsgHandle.RF.LocalOscillator.LOOutEnabled = true;
-                rfsgHandle.RF.LocalOscillator.Source = RfsgLocalOscillatorSource.Onboard;
-            }
-            else
-            {
-                rfsgHandle.RF.LocalOscillator.LOOutEnabled = false;
-                // Return to the default value, in case in future modifications the above case changes
-                // this to something other than the default
-                rfsgHandle.RF.LocalOscillator.Source = RfsgLocalOscillatorSource.Onboard;
+                switch (loConfig.SharingMode)
+                {
+                    case LocalOscillatorSharingMode.None:
+                        rfsgHandle.RF.LOOutExportConfigureFromRfsa = RfsgLOOutExportConfigureFromRfsa.Disabled;
+                        rfsgHandle.RF.LocalOscillator[loConfig.ChannelName].LOOutEnabled = false;
+                        rfsgHandle.RF.LocalOscillator[loConfig.ChannelName].Source = RfsgLocalOscillatorSource.Onboard;
+                        break;
+                    case LocalOscillatorSharingMode.Manual:
+                        rfsgHandle.RF.LOOutExportConfigureFromRfsa = RfsgLOOutExportConfigureFromRfsa.Disabled;
+                        rfsgHandle.RF.LocalOscillator[loConfig.ChannelName].LOOutEnabled = loConfig.ExportEnabled;
+                        rfsgHandle.RF.LocalOscillator[loConfig.ChannelName].Source = RfsgLocalOscillatorSource.FromString(loConfig.Source);
+                        break;
+                    default: // default to automatic case
+                        rfsgHandle.RF.LOOutExportConfigureFromRfsa = RfsgLOOutExportConfigureFromRfsa.Enabled;

```

Comment:
For automatic sharing for 5830/31, we also have to set the LO Source to `SG_SA_Shared`. This line here only matters for the 5840. How will we handle this within this case?